document-store/src/main/java/org/hypertrace/core/documentstore/expression/FilteringExpression.java

```diff
@@ -0,0 +1,11 @@
+package org.hypertrace.core.documentstore.expression;
+
+import org.hypertrace.core.documentstore.parser.FilteringExpressionParser;
+
+/**
+ * An interface to represent that the expression can be used in either the WHERE clause or the
```

Comment:
It might be cleaner to separate out the where and having expressions. Having clause is for aggregates. So, separating these two will make it less error-prone

Reply:
Even Aaron had given the same comment. Ref.: https://github.com/hypertrace/document-store/pull/58#discussion_r745134184.

1. Even after separating out HavingExpression separately, it would have an identical structure since databases like MongoDB support aliases (IdentifierExpression) to be used in the conditional statements (than just AggregateExpressions)
2. For databases with different HAVING and WHERE implementations (like Postgres), it can be separated out at the Parser hierarchy.

In summary, a filter is common despite its usage in WHERE/HAVING clauses. The differentiation can be done in the implementation (at the Parser hierarchy).